aspnetcore/grpc/browser.md

```diff
@@ -11,156 +11,22 @@ uid: grpc/browser
+gRPC on ASP.NET Core offers two browser compatible solutions:
 
-For instructions on creating a gRPC project, see <xref:tutorials/grpc/grpc-start>.
+* **gRPC-Web** allows browser apps to call gRPC services with the gRPC-Web client and Protobuf. gRPC-Web requires the browser app to generate a gRPC client. gRPC-Web allows browser apps to benefit from the high-performance and low network usage of gRPC.
 
-## gRPC-Web in ASP.NET Core vs. Envoy
+  .NET has built-in support for gRPC-Web. For more information, see <xref:grpc/grpcweb>.
 
-There are two choices for how to add gRPC-Web to an ASP.NET Core app:
+* **gRPC JSON transcoding** allows browser apps to call gRPC services as if they were RESTful APIs with JSON. The browser app doesn't need to generate a gRPC client or know anything about gRPC. RESTful APIs can be automatically created from gRPC services by annotating the `.proto` file with HTTP metadata. Transcoding allows an app to support both gRPC and JSON web APIs, without duplicating effort of building separate services for both.
 
-* Support gRPC-Web alongside gRPC HTTP/2 in ASP.NET Core. This option uses middleware provided by the `Grpc.AspNetCore.Web` package.
-* Use the [Envoy proxy's](https://www.envoyproxy.io/) gRPC-Web support to translate gRPC-Web to gRPC HTTP/2. The translated call is then forwarded onto the ASP.NET Core app.
-
-There are pros and cons to each approach. If an app's environment is already using Envoy as a proxy, it might make sense to also use Envoy to provide gRPC-Web support. For a basic solution for gRPC-Web that only requires ASP.NET Core, `Grpc.AspNetCore.Web` is a good choice.
-
-## Configure gRPC-Web in ASP.NET Core
-
-gRPC services hosted in ASP.NET Core can be configured to support gRPC-Web alongside HTTP/2 gRPC. gRPC-Web does not require any changes to services. The only modification is startup configuration.
-
-To enable gRPC-Web with an ASP.NET Core gRPC service:
-
-* Add a reference to the [Grpc.AspNetCore.Web](https://www.nuget.org/packages/Grpc.AspNetCore.Web) package.
-* Configure the app to use gRPC-Web by adding `UseGrpcWeb` and `EnableGrpcWeb` to `Startup.cs`:
-
-[!code-csharp[](~/grpc/browser/sample/Startup.cs?name=snippet_1&highlight=10,14)]
-
-The preceding code:
-
-* Adds the gRPC-Web middleware, `UseGrpcWeb`, after routing and before endpoints.
-* Specifies the `endpoints.MapGrpcService<GreeterService>()` method supports gRPC-Web with `EnableGrpcWeb`. 
-
-Alternatively, the gRPC-Web middleware can be configured so all services support gRPC-Web by default and `EnableGrpcWeb` isn't required. Specify `new GrpcWebOptions { DefaultEnabled = true }` when the middleware is added.
-
-[!code-csharp[](~/grpc/browser/sample/AllServicesSupportExample_Startup.cs?name=snippet_1&highlight=12)]
+  .NET has built-in support for creating JSON web APIs from gRPC services. For more information, see <xref:grpc/httpapi>.
```

Comment:
> I think this is going to 💥 **_AXEPLODE_** 💥 because the new topic is 7.0 or later but linked here from a 3.0 or later topic. It's isn't breaking here and now on this build, but I think we had some break on live merge builds.

Why does it work here but break in live builds? TBH I don't have a problem with older versions of pages linking to this .NET 7 content. I'd like for people to know that it's an option if they upgrade to .NET 7.

Reply:
For a Blazor doc, it broke&dagger; recently on ...

https://github.com/dotnet/AspNetCore.Docs/pull/26401

&dagger;"broke" ... It's an opinion of mine just based on user experience. When they follow the link from a <7.0 doc, it will load **_but show the 'this content isn't available for this version' banner_**, which isn't ideal imo. I wish we could add the `?view=aspnetcore-7.0` to the <7.0 versions of the content to avoid this happening, but the build system doesn't honor it when building the links.

I just fixed it for the Blazor doc in that case. We didn't address it for migration and What's New content going forward generally. Idk why the suggestion shows for a live build but not for a specific doc built on a PR. When my Blazor link suggestion showed up in the live build, it didn't break when the link was first placed IIRC, but I'd need to research to confirm it.

Reply:
I'm fine with a banner in that situation.

Reply:
AFAIK, there are only ~30 suggestions on this from the build system. If that ever goes to **_hundreds_** of suggestions, it might make the build report a PITA to use to find *real problems* before going live. That should only be a problem in the distant future tho ... long after I'm 💀 and ⚰️😆.